src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/scheduler/removesolicitorfromcase/RemoveSolicitorFromCaseTask.java

```diff
@@ -0,0 +1,110 @@
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.ccd.client.model.SearchResult;
+import uk.gov.hmcts.reform.ccd.client.model.StartEventResponse;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CaseType;
+import uk.gov.hmcts.reform.finrem.caseorchestration.service.CcdDataStoreService;
+import uk.gov.hmcts.reform.finrem.caseorchestration.service.CcdService;
+import uk.gov.hmcts.reform.finrem.caseorchestration.service.SystemUserService;
+
+import java.io.IOException;
+import java.util.List;
+
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.EventType.AMEND_CASE_CRON;
+
+/**
+ * Scheduled task to remove solicitor access from cases.
+ * To enable the task to execute set environment variables:
+ * <ul>
+ *     <li>CRON_REMOVE_SOLICITOR_FROM_CASE_ENABLED=true</li>
+ *     <li>TASK_NAME=RemoveSolicitorFromCaseTask</li>
+ * </ul>
+ * and add data to the <code>removeSolicitorFromCase.csv</code> file.
+ */
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class RemoveSolicitorFromCaseTask implements Runnable {
+
+    @Value("${cron.removeSolicitorFromCase.enabled:false}")
+    private boolean taskEnabled;
+
+    private static final String TASK_NAME = "RemoveSolicitorFromCaseTask";
+
+    private static final String FILENAME = "removeSolicitorFromCase.csv";
+
+    private final RemoveSolicitorFromCaseFileReader removeSolicitorFromCaseFileReader;
+    private final SystemUserService systemUserService;
+    private final CcdService ccdService;
+    private final CcdDataStoreService ccdDataStoreService;
+
+    @Override
+    public void run() {
+        log.info("Scheduled task {} enabled: {}", TASK_NAME, taskEnabled);
+        if (!taskEnabled) {
+            return;
+        }
+
+        List<RemoveSolicitorFromCaseRequest> requests;
+        try {
+            requests = removeSolicitorFromCaseFileReader.read(FILENAME);
+        } catch (IOException e) {
+            throw new RemoveSolicitorFromCaseException("Unable to read requests", e);
+        }
+        if (requests.isEmpty()) {
+            log.info("No requests found");
+            return;
+        }
+
+        String systemUserToken = getSystemUserToken();
+        requests.forEach(r -> handleRequest(r, systemUserToken));
+    }
+
+    private void handleRequest(RemoveSolicitorFromCaseRequest request, String systemUserToken) {
+        log.info("Removing {} access to case {} for solicitor {}", request.role(), request.caseReference(),
+            request.userId());
+
+        CaseType caseType = CaseType.forValue(request.caseType());
+        SearchResult searchResult = ccdService.getCaseByCaseId(request.caseReference(), caseType, systemUserToken);
+
+        if (searchResult.getTotal() != 1) {
```

Comment:
Can we ignore and log the request when the total equals to zero? therefore, it will not block the other requests.

Reply:
My thinking was that I wanted such an error to stop processing completely. If we didn't get exactly one result then most likely there is an error in the input file e.g. wrong case reference.
In reality there is probably only ever going to be 1 or possibly 2 requests whenever this is run.
I think in a scenario where we are doing batch processing of a large number of records then it definitely makes sense not to block other requests.

Despite all that I can change it.